Sources/Core/Schema.swift

```diff
@@ -242,14 +255,18 @@ extension Schema {
                     }))
                 }
             case JSONType.object:
+                let requiredProps = Set(propertyInfo["required"] as? [String] ?? [])
```

Comment:
Not for this PR, but as an idea for all of this stringly accessed property infos should we create an enum for that includes the available property info keys?